src/Test/StateMachine/Internal/Types.hs

```diff
@@ -49,16 +51,20 @@ import           Test.StateMachine.Types
 
 ------------------------------------------------------------------------
 
+-- | Type-level function that maybe returns a reference.
 type family MayResponse_ (refs :: TyFun ix k -> Type) (resp :: Response ix) :: k where
   MayResponse_ refs ('Response  t) = ()
   MayResponse_ refs ('Reference i) = refs @@ i
 
+-- | Internal untyped commands.
 data IntRefed (f :: Signature ix) where
   IntRefed :: ( Show     (Response_ ConstIntRef resp)
               , Typeable (Response_ ConstIntRef resp)
               , Typeable resp
               ) => f ConstIntRef resp -> MayResponse_ ConstIntRef resp -> IntRefed f
 
+-- | Forks are used to represent parallel programs. They have a sequential
+-- prefix, the middle argument, and two parallel branches.
```

Comment:
This reads as if they have three things:
* sequential prefix
* middle argument
* two parallel branches